python/semantic_kernel/connectors/memory/azure_cosmos_db/azure_cosmos_db_no_sql_store.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import sys
+from collections.abc import Sequence
+from typing import Any, TypeVar
+
+if sys.version_info >= (3, 12):
+    from typing import override  # pragma: no cover
+else:
+    from typing_extensions import override  # pragma: no cover
+
+from azure.cosmos.aio import CosmosClient
+
+from semantic_kernel.connectors.memory.azure_cosmos_db.azure_cosmos_db_no_sql_base import AzureCosmosDBNoSQLBase
+from semantic_kernel.connectors.memory.azure_cosmos_db.azure_cosmos_db_no_sql_collection import (
+    AzureCosmosDBNoSQLCollection,
+)
+from semantic_kernel.data.record_definition.vector_store_model_definition import VectorStoreRecordDefinition
+from semantic_kernel.data.vector_storage.vector_store import VectorStore
+from semantic_kernel.data.vector_storage.vector_store_record_collection import VectorStoreRecordCollection
+from semantic_kernel.exceptions.memory_connector_exceptions import MemoryConnectorException
+from semantic_kernel.utils.experimental_decorator import experimental_class
+
+TModel = TypeVar("TModel")
+
+
+@experimental_class
+class AzureCosmosDBNoSQLStore(AzureCosmosDBNoSQLBase, VectorStore):
+    """A VectorStore implementation that uses Azure CosmosDB NoSQL as the backend storage."""
+
+    def __init__(
+        self,
+        database_name: str,
+        url: str | None = None,
+        key: str | None = None,
+        cosmos_client: CosmosClient | None = None,
+        create_database: bool = False,
+    ):
+        """Initialize the AzureCosmosDBNoSQLStore.
+
+        Args:
+            database_name (str): The name of the database. The database may not exist yet.
+                                 If it does not exist, it will be created when the first collection is created.
+            url (str): The URL of the Azure Cosmos DB NoSQL account. Defaults to None.
+            key (str): The key of the Azure Cosmos DB NoSQL account. Defaults to None.
+            cosmos_client (CosmosClient): The custom Azure Cosmos DB NoSQL client whose lifetime is managed by the user.
+                                          Defaults to None.
+            create_database (bool): If True, the database will be created if it does not exist.
+                                    Defaults to False.
+        """
+        super().__init__(
```

Comment:
I am actually not sure if having a flag like this one in the base class is necessary. It seems like we can achieve the same thing without using the flag and it gives more flexibility to individual implementations.

Nonetheless, I added that to this PR. We can revisit this later.

Reply:
I don't think so, the whole point is that we don't close clients we have not created and so we need a way to distinguish between a client that was passed in (non-managed) and a client we create, once a client is created we don't know which one it was, hence we set the flag, the public interface doesn't see it. But let me know what you are thinking, we can still change it now!

Reply:
I agree. We should manage the client we create. However, I think the `managed_client` flag is the base class is uncessary because all derived classes now have to set it, which dictates how those implementations manage their clients. For instance, the cosmos db client is managed in a base class. An other example: the ai search requires two clients, which requires another flag that is not in the vector vase class.

Reply:
I have set it up as optional, so if not appropriate then it can be ignored, but I feel having this done in the same way for all (when needed) is beneficial, and in this case, I would argue that you need this flag as well in order to not close a client that was passed in? I see now that both have the same aexit methods, why not move that into the base class as well and replicate the same flag in it?

Reply:
I agree we should have some flag indicating whether we should clean up the client. However, I am not certain that having this flag in the base class of all the stores and collections is the proper way of doing so. I feel like giving that responsibility back to the individual implementations may allow more flexibility. We should leave as many options open as possible.

From a code readability perspective, we will now have this flag all over the place if I replicate it in the cosmos bass class.